From f119c0f83c856aeca0e8583edcd25425fe69a2da Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 30 May 2024 11:38:11 +0200
Subject: [PATCH 607/841] feat(fdts): add sdmmc2 node for stm32mp21

Add sdmmc2 node for stm32mp21.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 2d4228be2067fbf798eb8f829ee133433f16dbdc feat(fdts): add sdmmc2 node for stm32mp21]
Change-Id: I0362f45a79b5ac3bab6d09eba0e6b09602987b4e
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/384300
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
---
 fdts/stm32mp21-bl2.dtsi  |  3 ++-
 fdts/stm32mp21-bl31.dtsi |  1 +
 fdts/stm32mp211.dtsi     | 13 +++++++++++++
 3 files changed, 16 insertions(+), 1 deletion(-)

diff --git a/fdts/stm32mp21-bl2.dtsi b/fdts/stm32mp21-bl2.dtsi
index 325c551c3..82d56ec9f 100644
--- a/fdts/stm32mp21-bl2.dtsi
+++ b/fdts/stm32mp21-bl2.dtsi
@@ -7,9 +7,10 @@
 
 / {
 	soc@0 {
-#if !STM32MP_SDMMC
+#if !STM32MP_EMMC && !STM32MP_SDMMC
 		bus@42080000 {
 			/delete-node/ mmc@48220000;
+			/delete-node/ mmc@48230000;
 		};
 #endif
 	};
diff --git a/fdts/stm32mp21-bl31.dtsi b/fdts/stm32mp21-bl31.dtsi
index 4a2310375..b54c0e802 100644
--- a/fdts/stm32mp21-bl31.dtsi
+++ b/fdts/stm32mp21-bl31.dtsi
@@ -7,6 +7,7 @@
 	soc@0 {
 		bus@42080000 {
 			/delete-node/ mmc@48220000;
+			/delete-node/ mmc@48230000;
 		};
 	};
 };
diff --git a/fdts/stm32mp211.dtsi b/fdts/stm32mp211.dtsi
index d66fec410..a41a848c8 100644
--- a/fdts/stm32mp211.dtsi
+++ b/fdts/stm32mp211.dtsi
@@ -249,6 +249,19 @@
 				max-frequency = <120000000>;
 				status = "disabled";
 			};
+
+			sdmmc2: mmc@48230000 {
+				compatible = "st,stm32mp25-sdmmc2", "arm,pl18x", "arm,primecell";
+				arm,primecell-periphid = <0x00353180>;
+				reg = <0x48230000 0x400>, <0x44230800 0x8>;
+				clocks = <&rcc CK_KER_SDMMC2>;
+				clock-names = "apb_pclk";
+				resets = <&rcc SDMMC2_R>;
+				cap-sd-highspeed;
+				cap-mmc-highspeed;
+				max-frequency = <120000000>;
+				status = "disabled";
+			};
 		};
 
 		risaf2: risaf@420b0000 {
-- 
2.47.2

